rules/windows/defense_evasion_volume_shadow_copy_deletion_via_powershell.toml

```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2021/07/19"
+maturity = "production"
+updated_date = "2021/07/19"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies use of WMI’s Win32_ShadowCopy class or CIM cmdlets for shadow copy deletion on endpoints. This commonly occurs in tandem with ransomware or
+other destructive attacks.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Volume Shadow Copy Deletion via PowerShell"
+references = [
+    "https://docs.microsoft.com/en-us/previous-versions/windows/desktop/vsswmi/win32-shadowcopy",
+    "https://powershell.one/wmi/root/cimv2/win32_shadowcopy",
+    "https://www.fortinet.com/blog/threat-research/stomping-shadow-copies-a-second-look-into-deletion-methods",
+]
+risk_score = 73
+rule_id = "d99a037b-c8e2-47a5-97b9-170d076827c4"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+  process.name: ("powershell.exe", "pwsh.exe") and 
+  process.args : "Get-WmiObject" and process.args : ("Win32_ShadowCopy","% { $_.Delete() }", "Disabled") or
+  process.args : "Get-CimInstance" and process.args : ("Win32_ShadowCopy", "Remove-CimInstance")
```

Comment:
you might want to rework your `process.args` testing. If I make the command `powershell -c "Get-WmiObject Win32_ShadowCopy | % { $_.Delete() }"`, then process.args becomes `["powershell", "-c", "Get-WmiObject Win32_ShadowCopy | % { $_.Delete() }"]` and this wouldn't match.

Maybe a wildcard on command line would be better here?

Reply:
@rw-access  `*Delete()*`

Reply:
This would catch the gwmi (Get-WmiObject) and gcim (Get-CimInstance) aliases?

Reply:
yeah exactly. every term would need `*` around it. and at that point, you'd probably be better using command line and a single check

Reply:
@austinsonger there are some variants to this that could end up bypassing the query:

```
Get-WmiObject Win32_ShadowCopy | Remove-WmiObject
gwmi Win32_ShadowCopy | Remove-WmiObject
gwmi Win32_ShadowCopy | rwmi
gcim Win32_ShadowCopy | rcim
```